libfoundation/src/foundation-value.cpp

```diff
@@ -802,7 +802,7 @@ bool __MCValueInitialize(void)
 
 void __MCValueFinalize(void)
 {
-    for(uindex_t i = 0; i < sizeof(s_value_pools) / sizeof(s_value_pools[0]); i++)
+    for(index_t i = 0; i < sizeof(s_value_pools) / sizeof(s_value_pools[0]); i++)
```

Comment:
Because s_value_pools is a pointer, not an array, I suspect. I think the comparison is getting turned into i < 0, which is tautological for unsigned integers.